pkg/mempool/simplemempool/internal/parts/formbatches/formbatches.go

```diff
@@ -0,0 +1,95 @@
+package formbatches
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/filecoin-project/mir/pkg/dsl"
+	mpdsl "github.com/filecoin-project/mir/pkg/mempool/dsl"
+	"github.com/filecoin-project/mir/pkg/mempool/simplemempool/internal/common"
+	mppb "github.com/filecoin-project/mir/pkg/pb/mempoolpb"
+	"github.com/filecoin-project/mir/pkg/pb/requestpb"
+	t "github.com/filecoin-project/mir/pkg/types"
+)
+
+type State struct {
+	*common.State
+	NewTxIDs []t.TxID
+}
+
+// IncludeBatchCreation registers event handlers for processing new transactions and forming batches.
+func IncludeBatchCreation(
+	m dsl.Module,
+	mc *common.ModuleConfig,
+	params *common.ModuleParams,
+	commonState *common.State,
+) {
+	state := &State{
+		State:    commonState,
+		NewTxIDs: nil,
+	}
+
+	dsl.UponNewRequests(m, func(requests []*requestpb.Request) error {
+		txs := make([][]byte, len(requests))
+		for i, req := range requests {
+			var err error
+			txs[i], err = proto.Marshal(req)
```

Comment:
Protobuf marshalling is not guaranteed to be deterministic. If two nodes receive the same request, they might assign it different IDs in their respective mempools. Not 100% sure this is a correctness issue, but if nothing else, it might be confusing for someone trying to understand the handling of requests.
Custom (deterministic) serialization of a request would be nice here.

Reply:
Fixed